homeassistant/components/frigidaire/climate.py

```diff
@@ -0,0 +1,235 @@
+        convert = {
+            frigidaire.Mode.OFF.value: HVAC_MODE_OFF,
+            frigidaire.Mode.COOL.value: HVAC_MODE_COOL,
+            frigidaire.Mode.FAN.value: HVAC_MODE_FAN_ONLY,
+            frigidaire.Mode.ECO.value: HVAC_MODE_AUTO,
+        }
+
+        frigidaire_mode = self._details.for_code(
+            frigidaire.HaclCode.AC_MODE
+        ).number_value
+
+        return convert[frigidaire_mode]
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return (
+            self._details.for_code(frigidaire.HaclCode.AMBIENT_TEMPERATURE)
+            .containers.for_id(frigidaire.ContainerId.TEMPERATURE)
+            .number_value
+        )
+
+    @property
+    def fan_mode(self):
+        """Return the fan setting."""
+        convert = {
+            frigidaire.FanSpeed.OFF.value: FAN_OFF,  # when the AC is off
+            frigidaire.FanSpeed.AUTO.value: FAN_AUTO,
+            frigidaire.FanSpeed.LOW.value: FAN_LOW,
+            frigidaire.FanSpeed.MEDIUM.value: FAN_MEDIUM,
+            frigidaire.FanSpeed.HIGH.value: FAN_HIGH,
+        }
+        fan_speed = self._details.for_code(frigidaire.HaclCode.AC_FAN_SPEED_SETTING)
+
+        if not fan_speed:
+            return FAN_OFF
+
+        return convert[fan_speed.number_value]
+
+    @property
+    def min_temp(self):
+        """Return the minimum temperature."""
+        if self.temperature_unit == TEMP_FAHRENHEIT:
+            return 60
+
+        return 16
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature."""
+        if self.temperature_unit == TEMP_FAHRENHEIT:
+            return 90
+
+        return 32
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+        if temperature is None:
+            return
+        temperature = int(temperature)
+
+        self._client.execute_action(
+            self._appliance, frigidaire.Action.set_temperature(temperature)
+        )
+
+    def set_fan_mode(self, fan_mode):
+        """Set new target fan mode."""
+        convert = {
+            FAN_AUTO: frigidaire.FanSpeed.AUTO,
+            FAN_LOW: frigidaire.FanSpeed.LOW,
+            FAN_MEDIUM: frigidaire.FanSpeed.MEDIUM,
+            FAN_HIGH: frigidaire.FanSpeed.HIGH,
+        }
+
+        # Guard against unexpected fan modes
+        if fan_mode not in convert:
+            return
+
+        action = frigidaire.Action.set_fan_speed(convert[fan_mode])
+        self._client.execute_action(self._appliance, action)
+
+    def set_hvac_mode(self, hvac_mode):
+        """Set new target operation mode."""
+        if hvac_mode == HVAC_MODE_OFF:
+            self._client.execute_action(
+                self._appliance, frigidaire.Action.set_power(frigidaire.Power.OFF)
+            )
+            return
+
+        convert = {
+            HVAC_MODE_AUTO: frigidaire.Mode.ECO,
```

Comment:
HVAC_MODE_AUTO is used for when on a schedule, learned behavior or AI.
I don't think eco matches that?

I think this should probably be part of presets.